File: frontend/src/types/risk.ts
/** Materialized deterministic risks of a COMPLETED inspection (Stage 7). */

export type RiskSeverityValue = 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL';

export type RiskStatusValue = 'active' | 'resolved' | 'all';

export interface RiskRead {
  id: string;
  room_id: string;
  inspection_id: string;
  risk_code: string;
  rule_code: string;
  rule_version: number;
  severity: RiskSeverityValue;
  title_key: string;
  explanation_key: string;
  consequence_key: string;
  mitigation_key: string;
  communication_key: string;
  warranty_exclusion_candidate: boolean;
  blocks_finishing: boolean;
  source_signature: string;
  is_active: boolean;
  resolved_at: string | null;
  position: number | null;
  created_at: string;
  updated_at: string;
}

export interface RiskSourceFindingRead {
  finding_id: string | null;
  finding_key_snapshot: string;
  value_snapshot: Record<string, unknown> | null;
  position: number | null;
}

export interface RiskDetail extends RiskRead {
  source_findings: RiskSourceFindingRead[];
}

export interface RiskEvaluateRequest {
  inspection_id: string;
}

export interface RiskListResponse {
  items: RiskRead[];
  total: number;
}

export interface RiskDetailResponse {
  items: RiskDetail[];
  total: number;
}

File: backend/app/schemas/risk.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.risk import RiskSeverity


class RiskEvaluateRequest(BaseModel):
    """Request body for an idempotent risk evaluation of one inspection."""

    inspection_id: uuid.UUID


class RiskRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    room_id: uuid.UUID
    inspection_id: uuid.UUID
    risk_code: str
    rule_code: str
    rule_version: int
    severity: RiskSeverity
    title_key: str
    explanation_key: str
    consequence_key: str
    mitigation_key: str
    communication_key: str
    warranty_exclusion_candidate: bool
    blocks_finishing: bool
    source_signature: str
    is_active: bool
    resolved_at: datetime | None
    position: int | None
    created_at: datetime
    updated_at: datetime


class RiskSourceFindingRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    finding_id: uuid.UUID | None
    finding_key_snapshot: str
    value_snapshot: dict | None
    position: int | None


class RiskDetailRead(RiskRead):
    source_findings: list[RiskSourceFindingRead] = []


class RiskListResponse(BaseModel):
    items: list[RiskRead]
    total: int


class RiskDetailResponse(BaseModel):
    items: list[RiskDetailRead]
    total: int
